pycommon/ip_prefix_search.py

```diff
@@ -203,6 +207,16 @@ def __len__(self):
 
     @classmethod
     def fromFile(cls, val):
+        """ Initialize IPPSContext from blacklist data file. Function parse source file 
+        and create IPPSNetwork structs from each line. 
+        Blacklist file must be in format:
+	<ip address>/<mask>,<data>\n
+        
+        Args: 
+            val: path to source file
+        Return: 
+            new IPPSContext or None if val isn't string 
+        """
         if isinstance(val, str):
```

Comment:
This is not common in Python. Normal way is to don't check type of `val`, just try to use it in open(). When it fails, an exception explaining what is the problem and where is raised, which is good for debugging. (I assume it's not expected that a non-string will be passed to this function).

These's more similar cases throughout the code. In general, checking for a particular type is dicouraged in Python (mostly because of "duck-typing"). In most cases, it's better to just try to use the value (and catch exception if needed).